bundles/org.eclipse.e4.ui.workbench/src/org/eclipse/e4/ui/workbench/modeling/IModelProcessorContribution.java

```diff
@@ -59,8 +84,30 @@ public interface IModelProcessorContribution {
 	 * Component annotation:<br>
 	 * <code>@Component(property = { IModelProcessorContribution.APPLY_PROPERTY_PREFIX + "initial" })</code>
 	 * </p>
+	 *
+	 * @deprecated Instead annotate the component with the {@link Apply @Apply(&lt
+	 *             a-value &gt)} component property type
 	 */
+	@Deprecated(forRemoval = true, since = "1.16")
 	String APPLY_PROPERTY_PREFIX = "apply="; //$NON-NLS-1$
+
+	/**
+	 * An OSGi service component property type used to specify the {@code apply}
+	 * attribute, which defines in which case a processor is run. If not specified
+	 * it defaults to {@code always}.
+	 *
+	 * @since 1.16
+	 * @see #APPLY_PROPERTY_KEY
+	 * @see #APPLY_ALWAYS
+	 * @see #APPLY_INITIAL
+	 *
+	 */
+	@ComponentPropertyType
+	@Retention(RetentionPolicy.SOURCE)
+	@interface Apply {
+		boolean value();
```

Comment:
```suggestion
		String value() default "always";
```

Reply:
Indeed the return type is wrong, copy-paste mistake. Thanks for pointing this out.
The return type also makes sense.
Applied both.